### PR TITLE
Nav accordion no jquery

### PR DESCRIPTION
**Pull request by mattheu**
Updated to use vanilla JS only.

### CODE REVIEW

File: src/js/NavAccordion.js [hunk 1]
```diff
@@ -1,54 +1,71 @@
...
+			});
+
 		} );
 
 	}
 
-	$('.NavAccordion').each( function() {
-		navAccordion( $(this) );
-	} );
+	var els = document.getElementsByClassName('NavAccordion');
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: A graceful failure would be dandy here, at first glance `( 'getElementsByClassName' in document && 'addEventListener' in document  && 'classList' in document.documentElement )` should cover the basics.

File: src/js/NavAccordion.js [hunk 2]
```diff
@@ -1,54 +1,71 @@
-(function($) {
+(function() {
 
-	var navAccordion = function( $el ) {
+	var navAccordion = function( el ) {
 
-		$el.children().each( function() {
+		Array.prototype.forEach.call( el.children, function( navAccordion_Item ) {
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: CS: Naming convention for JS is camelCase

File: src/js/NavAccordion.js [hunk 3]
```diff
@@ -1,54 +1,71 @@
...
-						$NavAccordion_Item.removeClass( 'NavAccordion_Item-Open' );
-						$subNav.css( 'max-height', 0 );
-					}
+			if ( anchor.length ) {
+				anchor[0].appendChild( navAccordion_Toggle );
+			}
 
+			var toggleSubNav = function( show ) {
+
+				if ( 'undefined' === typeof( show ) ) {
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: `typeof show`

File: src/js/NavAccordion.js [hunk 4]
```diff
@@ -1,54 +1,71 @@
...
-		navAccordion( $(this) );
-	} );
+	var els = document.getElementsByClassName('NavAccordion');
+
+	for ( var i = 0; i < els.length; i++ ) {
+		navAccordion( els[ i ] );
+	}
 
-} )(jQuery);
+} )();
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: I'd pass window and document to avoid scope lookups, but not a blocker.

File: src/js/NavAccordion.js [hunk 5]
```diff
@@ -1,54 +1,71 @@
...
+				if ( 'undefined' === typeof( show ) ) {
+					show = navAccordion_Item.classList.contains( 'NavAccordion_Item-Closed' );
 				}
 
-				$NavAccordion_Toggle.appendTo( $NavAccordion_Item.children('.NavAccordion_Anchor') );
-				toggleSubNav( false );
+				if ( show ) {
+					navAccordion_Item.classList.remove( 'NavAccordion_Item-Closed' );
+					navAccordion_Item.classList.add( 'NavAccordion_Item-Open' );
+					subNav.style['max-height'] = openHeight + 'px';
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: `style.maxHeight` (same inside else too).

File: src/js/NavAccordion.js [hunk 6]
```diff
@@ -1,54 +1,71 @@
...
-				var $NavAccordion_Toggle = $( '<button/>', {
-					text: 'Toggle',
-					class: 'Btn NavAccordion_Toggle'
-				} );
+			openHeight += parseInt( style.marginTop ) + parseInt( style.marginBottom );
 
-				var toggleSubNav = function( show ) {
+			var navAccordion_Toggle = document.createElement( 'BUTTON' );
+			navAccordion_Toggle.appendChild( document.createTextNode( "Toggle" ) );
+			navAccordion_Toggle.classList.add( 'Btn' );
```
Comment (peterwilsoncc) [inline src/js/NavAccordion.js]: `classList.add( 'Btn', 'NavAccordion_Toggle' )` will add these all in one hit.

**Comment by peterwilsoncc** [comment]
Oh, misread this as assignecd to me and added a few notes.

File: src/js/NavAccordion.js (hunk 6 above)
Comment (mattheu) [inline src/js/NavAccordion.js]: [CanIUse.com](http://caniuse.com/#search=classlist) suggest that IE11 doesn't support multiple params being passed to `classList.add`

**Comment by mattheu** [comment]
Thanks for the review!

**Comment by sambulance** [comment]
@peterwilsoncc I think you're better at reviewing JS than me anyway!

File: src/js/NavAccordion.js (hunk 1 above)
Comment (mattheu) [inline src/js/NavAccordion.js]: I've added a check to the top of the file that will simply return if any of these methods are NOT supported.

**Comment by mattheu** [comment]
I'm going to go ahead and merge this into https://github.com/humanmade/hm-pattern-library/pull/41 - just confusing having so many PR's open. If you have any follow up - post it there.

**Pull request merged**